### PR TITLE
Warn for ranks with 0 elements

### PR DESCRIPTION
**Pull request by timofeymukha**

### CODE REVIEW

File: src/mesh/mesh.f90 [hunk 1]
```diff
@@ -188,6 +188,10 @@ subroutine mesh_init_nelv(this, gdim, nelv)
 
     call this%free()
 
+    if (this%nelv < 1) then
+       call neko_warning("One of the MPI ranks has 0 mesh elements!")
```
Comment (MartinKarp) [inline src/mesh/mesh.f90:192]: Could we perhaps also print the number of the rank?

**Review by MartinKarp — CHANGES_REQUESTED** [review]

File: src/mesh/mesh.f90 [hunk 2]
```diff
@@ -211,6 +215,9 @@ subroutine mesh_init_dist(this, gdim, dist)
     call this%free()
 
     this%nelv = dist%num_local()
+    if (this%nelv < 1) then
+       call neko_warning("One of the MPI ranks has 0 mesh elements!")
+    end if
```
Comment (njansson) [inline src/mesh/mesh.f90:220]:
```suggestion
    if (this%nelv < 1) then
       call neko_warning("Rank " // pe_rank // " ranks has zero mesh elements!")
    end if
```

@MartinKarp How about something like this?

Comment (timofeymukha) [reply src/mesh/mesh.f90:220]: I didnt know you could concat numbers into a string like that!

**Comment by adperezm** [comment]
Quick question, In which case could this happen? I only see this happening if nelv < mpiranks, or?


Am I missing something?

**Comment by timofeymukha** [comment]
> Quick question, In which case could this happen? I only see this happening if global_nelv < mpiranks, or?
> 
> Am I missing something?

Yes, like if you run the RB example on >25 ranks. So, in principle, only you attempt to run a small problem on a huge number of ranks.

File: src/mesh/mesh.f90 (hunk 2 above)
Comment (timofeymukha) [reply src/mesh/mesh.f90]: @njansson OK, it seems one actiually cannot :-)

**Review by MartinKarp — APPROVED** [review]

**Comment by timofeymukha** [comment]
Wait, some rogue files got in.. let me fix that :D

**Comment by timofeymukha** [comment]
There we go. I think VS auto-commits all changes when I push some button..

**Review by njansson — APPROVED** [review]

**Pull request merged**